github/Requester.py

```diff
@@ -834,7 +834,11 @@ def createException(
             exc = GithubException.BadUserAgentException
         elif status == 403 and cls.isRateLimitError(lc_message):
             exc = GithubException.RateLimitExceededException
-        elif status == 404 and (lc_message == "not found" or "no object found" in lc_message):
+        elif status == 404 and (
+            lc_message == "not found"  #
+            or lc_message == "branch not found"
```

Comment:
what about
```suggestion
            or " not found" in lc_message
```

Reply:
@EnricoMi Should I remove the check for message completely? This seems unnecessary if status code is 404. 

Can add tests 👍